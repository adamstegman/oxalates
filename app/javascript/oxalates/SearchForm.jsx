import React from 'react';

import './SearchForm.scss';

export class SearchForm extends React.Component {
  onSubmit(event) {
    event.preventDefault();
    this._onSubmit(event.target.value);
  }

  render() {
    const {
      query,
      onSubmit,
    } = this.props;
    this._onSubmit = onSubmit;

    const onChange = this.onSubmit.bind(this);
    return (
      <form className="search" onSubmit={onChange}>
        <label htmlFor="search">Search</label>
        <input type="search" name="search" placeholder="Search" value={query} onChange={onChange} />
        <input type="submit" value="Search" />
      </form>
    );
  }
};

SearchForm.propTypes = {
};
