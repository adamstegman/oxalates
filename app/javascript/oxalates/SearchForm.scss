@import "./actions-form";

$search_width: 9em;

.search {
  @extend %actions-form;
  min-width: $search_width;
}
