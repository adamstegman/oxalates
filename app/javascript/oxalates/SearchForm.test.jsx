import React from 'react';
import renderer from 'react-test-renderer';

import { SearchForm } from './SearchForm';

test('SearchForm renders the search form', () => {
  const component = renderer.create(
    <SearchForm />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SearchForm renders the search form with a query already entered', () => {
  const query = "my query"
  const component = renderer.create(
    <SearchForm query={query} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
