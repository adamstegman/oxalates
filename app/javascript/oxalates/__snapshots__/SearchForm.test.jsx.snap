// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders the search form 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <label
    htmlFor="search"
  >
    Search
  </label>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="search"
  />
  <input
    type="submit"
    value="Search"
  />
</form>
`;

exports[`SearchForm renders the search form with a query already entered 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <label
    htmlFor="search"
  >
    Search
  </label>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value="my query"
  />
  <input
    type="submit"
    value="Search"
  />
</form>
`;
